app/controllers/admin/themes_controller.rb

```diff
@@ -0,0 +1,56 @@
+class Admin::ThemesController < AdminController
+  before_action :set_theme, only: [:edit, :update, :destroy]
+
+  def index
+    @themes = policy_scope(Theme).order("LOWER(name)")
+  end
+
+  def new
+    authorize Theme
+
+    @theme = Theme.new
+  end
+
+  def edit; end
+
+  def create
+    authorize Theme
+
+    @theme = Theme.new(theme_params)
+
+    if @theme.save
+      redirect_to admin_themes_path
+    else
+      render :new
+    end
+  end
+
+  def update
```

Comment:
@sameera207 for update (& edit / delete) normally you would authorise the instance (may not matter atm depending on the Policy, but might still be nicer)

e.g.
authorize @ theme, :update?

Reply:
@craighawkes , thanks, I was more thinking, since edit, update and destroy first go through the `set_theme` method it will error out if there is a unauthorised access

Reply:
@sameera207 yep - your right sorry I missed that

maybe you don't need the 
authorize Theme
in update at all then (as its a double up)?

Reply:
@craighawkes , yes you are correct, no need to have `authorize Theme` in edit action, will fix that, thanks